### PR TITLE
Convert RDPFastPathPDU from an object to a Uint8Array

### PR DESCRIPTION
**Pull request by ibeckermayer**
in order to make use of cost-free js-rustwasm interop. See https://rustwasm.github.io/docs/wasm-bindgen/reference/types/number-slices.html#number-slices-u8-i8-u16-i16-u32-i32-u64-i64-f32-and-f64

### CODE REVIEW

**Comment by ibeckermayer** [comment]
https://github.com/gravitational/teleport/pull/35872#discussion_r1430819400

**Review by probakowski — APPROVED** [review]
LGTM, do we want to handle https://github.com/gravitational/teleport/pull/35872#discussion_r1430819400 here or in separate PR?

**Comment by zmb3** [comment]
My comments from the original still stand: https://github.com/gravitational/teleport/pull/35872#discussion_r1430819400

I don't mind if we address it here or in another PR, but I do feel strongly that we need to address them. We're somewhat protected by nature of websocket message boundaries, but skipping length checks when parsing binary protocols leads to all sorts of problems.

**Comment by ibeckermayer** [comment]
> My comments from the original still stand: [#35872 (comment)](https://github.com/gravitational/teleport/pull/35872#discussion_r1430819400)
> 
> I don't mind if we address it here or in another PR, but I do feel strongly that we need to address them. We're somewhat protected by nature of websocket message boundaries, but skipping length checks when parsing binary protocols leads to all sorts of problems.

For sure, I basically didn't know exactly what I was doing when I first wrote the code in that style and just went with it because it seemed to work. As a bonus of taking a second glance at this, it turns out that we avoid an unnecessary copy by favoring `new Uint8array(buffer, offset, length)` over anything with `buffer.slice()`.

3c43ff4bb9581c0e32aad23aef5077dd81b0077b

File: web/packages/teleport/src/lib/tdp/codec.ts [hunk 1]
```diff
@@ -952,12 +960,13 @@ export default class Codec {
...
-  decodeRDPFastPathPDU(buffer: ArrayBuffer): RDPFastPathPDU {
+  decodeRDPFastPathPDU(buffer: ArrayBuffer): RdpFastPathPdu {
+    const dv = new DataView(buffer);
     let offset = 0;
     offset += byteLength; // eat message type
+    const dataLength = dv.getUint32(offset);
     offset += uint32Length; // eat data_length
-    const data = buffer.slice(offset);
-    return new RDPFastPathPDU(new Uint8Array(data));
+    return new Uint8Array(buffer, offset, dataLength);
```
Comment (zmb3) [inline web/packages/teleport/src/lib/tdp/codec.ts:969]: Does `new Uint8Array` allocate new memory and copy into it? If so, we've lost the advantage of zero copy from wasm memory, right?

Is there a way to slice into the existing `ArrayBuffer` without reallocating?

Comment (zmb3) [reply web/packages/teleport/src/lib/tdp/codec.ts:969]: Edit: looks like `Uint8Array`'s constructor _DOES NOT_ copy when provided an existing buffer, so this should be good!

**Review by zmb3 — APPROVED** [review]

**Pull request merged**